### ISSUE TITLE
🐛 CSS parser failing on custom-variant syntax with Tailwind Directives enabled

### ISSUE THREAD

**Issue by Scriptornix** [biomejs/biome#9038]
### Environment information

<details>

```bash

CLI:
  Version:                      2.3.6
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.18.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0

```

</details>

### What happened?

In the playground, adding the following custom variant to a CSS file with Tailwind v4 causes an error in the parsing logic:

```css
@import "tailwindcss";

@custom-variant phx-click-loading (.phx-click-loading&, .phx-click-loading &);
```

```
parse: 
Expected `,` but instead found `&`
```

However, this is valid Tailwind v4 syntax.

### Expected result

The CSS parser should not error on valid Tailwind v4 syntax for custom-variant with a trailing ampersand.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
This is a bit of a weird one. `.phx-click-loading&` is not a valid css selector. However, it _is_ valid in tailwind specifically, where `&` gets replaced with the actual selector you’re extending.

**Comment by Scriptornix** [comment]
> This is a bit of a weird one. `.phx-click-loading&` is not a valid css selector. However, it _is_ valid in tailwind specifically, where `&` gets replaced with the actual selector you’re extending.

Ah gross, almost like Tailwind needs to be a separate language then instead of extensions to CSS. Do you happen to know if this syntax unique to the `@custom-variant` rule? I tried diving into the Tailwind docs to find more information about it to no avail.

In the meanwhile I replaced it with this syntax (which based on my testing is functionality equivalent, but I am by no means a tailwind expert):

```css
@import "tailwindcss";

@custom-variant phx-click-loading {
  :where(.phx-click-loading) &, 
  &:where(.phx-click-loading) {
    @slot;
  }
}
```

Cheers
